### PR TITLE
#158946377: Whitelist CORS origin for staging domain name 

### PR DESCRIPTION
**Pull request by cjmash**
**What does this PR do?**

- Adds the frontend  staging host name to  CORS_ORIGIN_WHITELIST to enable cross-site requests

**Description of Task to be completed?**

- Whitelist frontend staging domain name 

- Create  cors origin  regex 

**Relevant Pivotal Tracker stories**

[#158946377](https://www.pivotaltracker.com/story/show/158946377)

### CODE REVIEW

File: settings/prod.py [hunk 1]
```diff
@@ -6,8 +6,9 @@
                  'art-backend.herokuapp.com']
 
 CORS_ORIGIN_WHITELIST = (
-    'art-dashboard-staging.herokuapp.com'
+    'art-dashboard-staging.herokuapp.com',
+    'staging-art.andela.com'
 )
```
Comment (egichuri) [inline settings/prod.py]: we can exclude the staging ones here

File: settings/dev.py [hunk 2]
```diff
@@ -10,7 +10,18 @@
...
+                 '127.0.0.1']
+
+CORS_ORIGIN_WHITELIST = (
+    'art-dashboard-staging.herokuapp.com',
+    '127.0.0.1',
+    'staging-art.andela.com'
+)
+CORS_ORIGIN_REGEX_WHITELIST = \
+    (r'^(https?:\/\/)?(\w+\.)?((andela.com))?((localhost)\
+     |(127\.0\.0\.1)):\d{4}', )
```
Comment (egichuri) [inline settings/dev.py]: let's change this to 
```
CORS_ORIGIN_REGEX_WHITELIST = (
    r'^(https?:\/\/)?(\w+\.)?(andela.com)|((localhost)|(127\.0\.0\.1):\d{4})',
)
```

File: settings/prod.py (hunk 1 above)
Comment (cjmash) [reply settings/prod.py]: @egichuri  should I remove even the art heroku staging?

Comment (egichuri) [reply settings/prod.py]: yeah.. let's remove everything staging from prod.py

File: settings/dev.py [hunk 3]
```diff
@@ -10,7 +10,19 @@
...
-ALLOWED_HOSTS = [os.environ.get('MY_HOST_IP'), 'api-staging-art.andela.com']
+
+ALLOWED_HOSTS = [os.environ.get('MY_HOST_IP'), 'api-staging-art.andela.com',
+                 '127.0.0.1']
+
+CORS_ORIGIN_WHITELIST = (
+    'art-dashboard-staging.herokuapp.com',
+    '127.0.0.1',
+    'staging-art.andela.com'
+)
```
Comment (egichuri) [inline settings/dev.py]: the regex takes care of '127.0.0.1' and 'staging-art.andela.com'

Comment (cjmash) [reply settings/dev.py]: So there is no need to have them in the CORS_ORIGIN_WHITELIST? @egichuri

Comment (egichuri) [reply settings/dev.py]: yeah.. we can have only 'art-dashboard-staging.herokuapp.com' there

**Pull request merged**